### PR TITLE
Tweaks to the snakebox

### PR DESCRIPTION
**Pull request by Cyberboss**

### CODE REVIEW

**Comment by Incoming5643** [comment]
I very highly doubt that that sound is licensed in such a way that we can use it.

File: code/game/objects/structures/crates_lockers/closets/cardboardbox.dm [hunk 1]
```diff
@@ -30,20 +30,23 @@
...
 			break
 		if(Snake)
-			var/list/alerted = viewers(7,src)
-			if(alerted)
-				for(var/mob/living/L in alerted)
-					if(!L.stat)
-						L.do_alert_animation(L)
-						egged = 1
-				alerted << sound('sound/machines/chime.ogg')
+			alerted = viewers(7,src)
```
Comment (Incoming5643) [inline code/game/objects/structures/crates_lockers/closets/cardboardbox.dm]: This would alert both corpses and ghosts.

**Comment by AnturK** [comment]
Yeah, not sure on that. Source and licensing. It is a simple sound so might not be an issue but i'd rather be sure.

**Comment by Iamgoofball** [comment]
we could claim this as parody act, since it is indeed a parody

**Comment by Incoming5643** [comment]
That doesn't justify using the EXACT sound though. New age konami even feels like the kind of soulless corporation that'd D&C this sort of stuff too.

If I thought I could get away with _the sound_ I would have when I implemented this egg.

**Comment by Cyberboss** [comment]
If anyone can find a version that isn't exact I'll add it in. Otherwise, I guess I'll just revert it and add the small tweaks

File: code/__DEFINES/misc.dm [hunk 2]
```diff
@@ -564,4 +564,6 @@ var/global/list/ghost_others_options = list(GHOST_OTHERS_SIMPLE, GHOST_OTHERS_DE
 
 #define NUKE_NEAR_MISS 1
 #define NUKE_MISS_STATION 2
-#define NUKE_SYNDICATE_BASE 3
\ No newline at end of file
+#define NUKE_SYNDICATE_BASE 3
+
+#define SNAKE_SPAM_TICKS 600 //how long between cardboard box openings that trigger the '!'
```
Comment (Jordie0608) [inline code/__DEFINES/misc.dm]: I feel this doesn't really need to be a define when it's used once in a single file.

**Comment by Iamgoofball** [comment]
Actually parody act does justify using the exact sound

**Comment by Cyberboss** [comment]
How tf did that first commit pass?

File: code/game/objects/structures/crates_lockers/closets/cardboardbox.dm [hunk 3]
```diff
@@ -30,20 +31,23 @@
 /obj/structure/closet/cardboard/open()
 	if(opened || !can_open())
 		return 0
-	if(!egged)
+	var/list/alerted = null
+	if(world.time - egged > SNAKE_SPAM_TICKS)
```
Comment (ChangelingRain) [inline code/game/objects/structures/crates_lockers/closets/cardboardbox.dm]: this is a very nonstandard way to do a world.time cooldown, the standard way would be;

    if(egged < world.time) //or world.time > egged but that's just style choice
    	stuff
    egged = world.time + SNAKE_SPAM_TICKS

**Comment by optimumtact** [comment]
can someone clarify the situation on the sound ?

**Comment by Incoming5643** [comment]
Essentially it boils down to "do you think Konami would cease and desist us for using the most iconic noise in the metal gear series?"

My gut is leaning towards "Konami is terrible, so yes."

**Comment by Iamgoofball** [comment]
Fair use though

**Comment by Cheridan** [comment]
Parody only applies if you're using it to make a comment on the source material, like a political cartoon or something. The only comment this is making is "i like metal gear."

So no.

**Comment by Cyberboss** [comment]
Reverted the sound

**Pull request merged**